### PR TITLE
Implement Forgot Password Flow

### LINKED ISSUE #24: Forgot Password Flow
**Description**

Users need to be able to reset a password in the case they forgot their password. 

**Acceptance Criteria**
- As a user, I can enter my email address to receive an email from the backend
- Clicking that link takes me to a page where I can enter a new password
- I then log in with that new password

### PR DESCRIPTION
**Pull request by onb6**
## Why

#24 
Implemented the forgot password password feature to allow users to reset their password if they forget it by requesting a reset and following a link that is sent to them via email. 

* A few more changes need to be made to clean up some small things, but general workflow is implemented *

## This PR

- Created the page that displays when a user clicks the the "Forgot Password" link on the login screen.
- Added forgotPassword to the ApiClient and created a post request to the API forgot_password/request endpoint.
- Created the landing page from the email for the user to enter and confirm their new password. 
- Added forgotPasswordReset to the ApiClient and created a post request to the API forgot_password/reset endpoint, using the secret key from the email link.

## Screenshots

Screen Shot 2021-02-17 at 12 00 26 PM (2)
Screen Shot 2021-02-17 at 11 58 58 AM (2)

## Verification Steps

The changes were verified by running the application and entering an email that is associated with a user that is a main contact. Then clicked the link in the email that was received and reset password using the form. Confirmed that logging in with the new password was successful. Used developer tools to monitor network traffic and ensure that requests were made correctly and received successful responses.

### CODE REVIEW

File: src/App.tsx
Comment (jackblanc) [inline src/App.tsx]: I think I deleted these files, you can just delete these lines

File: src/App.tsx
Comment (jackblanc) [inline src/App.tsx]: same here

File: src/App.tsx
Comment (jackblanc) [inline src/App.tsx]: 👍

File: src/auth/authClient.tsx
Comment (jackblanc) [inline src/auth/authClient.tsx]: We can do that based on the promise being rejected or resolved.

File: src/auth/authClient.tsx
Comment (jackblanc) [inline src/auth/authClient.tsx:73]: Do either of these routes return any data? Doesn't seem like it... 

You can chain .then.catch onto these functions _where you call them_ to alert the user.

File: src/containers/forgot-password/ForgotPassword.tsx
Comment (jackblanc) [inline src/containers/forgot-password/ForgotPassword.tsx]: So like, I definitely started the trend of just putting any for onFinish, but like, can we not :) 

Would it be correct to just make `any` -> `ForgotPasswordRequest`? That may type check easily...

File: src/containers/forgot-password/ForgotPasswordReset.tsx
Comment (jackblanc) [inline src/containers/forgot-password/ForgotPasswordReset.tsx]: What has to change? If you're gonna commit a comment, at least make it clear enough someone else could do the work if necessary.

**Review by jackblanc — COMMENTED** [review]
Incredible work here! Left a few questions but I'm very happy with how this turned out, nicely done. Definitely would appreciate if you could do the secret keys for email verification as well, pretty similar but requires backend changes too (emails aren't being sent).

File: src/containers/forgot-password/ForgotPasswordReset.tsx
Comment (rymaju) [inline src/containers/forgot-password/ForgotPasswordReset.tsx]: Are we checking that both passwords entered are the same?

**Review by rymaju — COMMENTED** [review]
I dont have many other comments that Jack hasn't already mentioned. Make the changes proposed and you should be good to go!

File: src/containers/forgot-password/ForgotPasswordReset.tsx
Comment (jackblanc) [reply src/containers/forgot-password/ForgotPasswordReset.tsx]: Ah we should. I just added that in `Settings`

Also just noticed this, but please rename to `containers/forgotPassword/index.tsx`

File: src/containers/forgot-password/ForgotPasswordReset.tsx
Comment (vrautela) [reply src/containers/forgot-password/ForgotPasswordReset.tsx]: Whoops, nothing has to change. We forgot to delete that ;)

File: src/App.tsx
Comment (jackblanc) [inline src/App.tsx]:
```suggestion
import ForgotPassword from './containers/forgotPassword';
```
If you rename the directory and change the file name to `index.tsx` you can import with this shortened file path.

File: src/App.tsx
Comment (jackblanc) [inline src/App.tsx]: I added an enum `Routes` in a recent update, so you can add this string to that then reference the enum wherever the string is used.

File: src/containers/forgot-password/ForgotPassword.tsx
Comment (jackblanc) [inline src/containers/forgot-password/ForgotPassword.tsx]: We should probably use an AntD alert for this if possible

**Review by jackblanc — COMMENTED** [review]
Few things to update based on changes I merged, let me know if you want to pair 👍

File: src/containers/forgot-password/ForgotPassword.tsx
Comment (vrautela) [reply src/containers/forgot-password/ForgotPassword.tsx]: We didn't include this in our most recent commit bc we figured that people might not necessarily want to have an alert. An alternative would just be displaying some text on screen or something

File: src/containers/forgot-password-reset/index.tsx
Comment (rymaju) [inline src/containers/forgot-password-reset/index.tsx]: does eslint complain about `alert`? I think it might be best to just navigate to a barebones "You successfully reset your password!" page, or swap out the rendered JSX to say something similar.

File: src/containers/forgot-password/index.tsx
Comment (rymaju) [inline src/containers/forgot-password/index.tsx]: Same as the previous comment, would be cool to render something instead of an alert

**Review by rymaju — APPROVED** [review]
I think that while we want to do something instead of just alert, I think this passes my standards for now. We can write another small ticket for rendering something instead. Wait for @jackblanc before merge though.

**Pull request merged**
